### PR TITLE
Update Travis setup

### PR DESCRIPTION
**Pull request by michaelbaudino**
This PR updates Travis setup to:
* use the more recent and faster Linux virtualized infrastructure (rather than the older containerized infrastructure) by switching Linux testing to Ubuntu Trusty
* enable testing on MacOS X
* use the latest patch version of each Ruby minor version, by declaring only the major and minor versions (RVM will assume the patch version is the latest available) (_e.g._ we use `2.4`, rather than `2.4.3`)
* prevent using Bundler 2, since this version does not support Ruby < 2.3 (while we do support Ruby 2.1 and 2.2)

### CODE REVIEW

**Comment by michaelbaudino** [comment]
@thibaultdalban The test fail on this PR with `Gemfile` due to `i18n` recent versions incompatibility. I think it should be merged anyway, it will offer a better overview of what's working and what's not to fix the former incompatibility.

File: .travis.yml [hunk 1]
```diff
@@ -1,13 +1,15 @@
-sudo: required
-dist: trusty
```
Comment (tjouan) [inline .travis.yml]: > use the more recent and faster Linux virtualized infrastructure (rather than the older containerized infrastructure) by switching Linux testing to Ubuntu Trusty

We already switched to Ubuntu Trusty since e2966a4f8.

Now about the change of infrastructure it doesn't seem effective.
Before this change:
```
Worker information
hostname: d34354d8-c92b-4ea1-abfc-be17633445f7@1.production-1-worker-org-gce-x71t
version: v6.1.0 https://github.com/travis-ci/worker/tree/a6071dd64658ecf13ec3190bfb9e9ca3c8f3e10d
instance: travis-job-f9799a07-703b-45ee-9ff2-a4cfcd2888cc travis-ci-garnet-trusty-1512502259-986baf0 (via amqp)

Operating System Details
Distributor ID:	Ubuntu
Description:	Ubuntu 14.04.5 LTS
Release:	14.04
Codename:	trusty
```
(gce is "google cloud engine")

After this change:
```
Worker information
hostname: 9923f557-b415-4c62-a739-50504b2abf4b@1.production-1-worker-org-gce-s54k
version: v6.1.0 https://github.com/travis-ci/worker/tree/a6071dd64658ecf13ec3190bfb9e9ca3c8f3e10d
instance: travis-job-8ce90d97-dbda-4dc1-8a51-8b9b0555602a travis-ci-garnet-trusty-1512502259-986baf0 (via amqp)

Operating System Details
Distributor ID:	Ubuntu
Description:	Ubuntu 14.04.5 LTS
Release:	14.04
Codename:	trusty
```

So maybe the change is OK, but we're in fact just removing "dead configuration"
which isn't necessary anymore?

File: .travis.yml [hunk 2]
```diff
@@ -1,13 +1,15 @@
...
+os:
+  - linux
+  - osx
+language: ruby
```
Comment (tjouan) [inline .travis.yml]: This doesn't seem related to your PR, or necessary.

File: .travis.yml [hunk 3]
```diff
@@ -1,13 +1,15 @@
...
-  - 2.5.0
-  - 2.4.3
-  - 2.3.6
-  - 2.2.9
-  - 2.1.10
+  - 2.5
+  - 2.4
+  - 2.3
+  - 2.2
+  - 2.1
```
Comment (tjouan) [inline .travis.yml]: Unfortunately, we specify the full version on "purpose" to avoid issues with
either Travis or RVM not using the last release for each branch :-(

2.5: built with 2.5.1, should be 2.5.3
2.4: built with 2.4.4, should be 2.4.5

It means we do have to specify and regularly update the "teeny" version number.
Unless we find a working solution of course.

Please see #213 which contain interesting info and feedback about this issue,
but also a little about the Travis infrastructure selection.

**Comment by michaelbaudino** [comment]
Closing because we split this PR into multiple isolated ones: #266, #267, #268, #269 and #270
